### DISCUSSION TITLE
VMWare images are huge

### DISCUSSION THREAD

**Discussion by nesies** [vmware/packer-examples-for-vsphere#976] [Ideas]
On VMWare, the template/OVF images are X0GB even if the OS takes only several GB.

If the feature exists on _packer_, it would be great to use it.

(if i export an OVA and then import it, the result is a less than 3GB virtual machine)

**Comment by tenthirtyam** [comment]
Do you have thin provisioned set?

**Reply by nesies** [reply]
yes thin is set.
During the installation, the RPMs are downloaded and then deleted, which seems is not reclaimed
(datastore on NFS)

**Reply by tenthirtyam** [reply]
NFS is thin-provisioned by nature, and the VMDKs will grow as needed based on usage up to the capacity defined for the virtual disks. You could add a step using dd to zero-fill the unused space, but the template size will remain the same as the max size consumed during the build.

You could also add the use of the `vsphere-template` post-processor to save it as a new template wish would reduce the size of the final
image since it would save a new template and can then delete the build source if you like.
